Add blog post for go.opentelemetry.io switch

damemi opened the pull request:
- Fixes #5086
- **Preview**: https://deploy-preview-5087--opentelemetry.netlify.app/blog/2024/go-opentelemetry-io/

dmathieu commented on content/en/blog/2024/go-opentelemetry-io-switch.md:
```diff
@@ -0,0 +1,67 @@
+In the future, we are also considering refactoring the go.opentelemetry.io app
+(or using a different approach entirely to serve requests). There is more info
+about this in the GitHub issue linked above. However, this is not part of the
+current transition and any further changes to go.opentelemetry.io will be
```
Do we want to communicate every change?

```suggestion
current transition and any possibly breaking changes to go.opentelemetry.io will be
```

dmathieu reviewed (commented):
Should we invite folks to reach out synchronously (in a slack channel) if they encounter issues?

chalin commented on content/en/blog/2024/go-opentelemetry-io-switch.md:
```diff
@@ -0,0 +1,67 @@
+---
+title: Announcing Planned Migration for go.opentelemetry.io
+linkTitle: Planned migration for go.opentelemetry.io
```
How about just:

```suggestion
title: Planned migration of go.opentelemetry.io
```

chalin commented on content/en/blog/2024/go-opentelemetry-io-switch.md:
```diff
@@ -0,0 +1,67 @@
+title: Announcing Planned Migration for go.opentelemetry.io
+linkTitle: Planned migration for go.opentelemetry.io
+date: 2024-08-22
+author:
+  >- # If you have only one author, then add the single name on this line in quotes.
+  [Mike Dame](https://github.com/damemi) (Google),
```
Please remove the comments from the front matter. E.g.:

```suggestion
author: '[Mike Dame](https://github.com/damemi) (Google)'
```

chalin commented on content/en/blog/2024/go-opentelemetry-io-switch.md:
```diff
@@ -0,0 +1,67 @@
+
+With contributions from Tyler Yahn and Austin Parker.
+
+## tl;dr
```
```suggestion
## TL;DR
```

chalin commented on content/en/blog/2024/go-opentelemetry-io-switch.md:
```diff
@@ -0,0 +1,67 @@
+
+## tl;dr
+
+**The app that serves requests to go.opentelemetry.io will be migrating to a new
+host on [DATE]. There are no changes required from you. We are making this
+announcement because it is possible, but unlikely, that there may be downtime
+during this transition.**
```
Let's avoid the bold here for the entire paragraph.

```suggestion
The app that serves requests to go.opentelemetry.io will be migrating to a new
host on [DATE]. There are no changes required from you. We are making this
announcement because it is possible, but unlikely, that there may be downtime
during this transition.
```

chalin commented on content/en/blog/2024/go-opentelemetry-io-switch.md:
```diff
@@ -0,0 +1,67 @@
+host on [DATE]. There are no changes required from you. We are making this
+announcement because it is possible, but unlikely, that there may be downtime
+during this transition.**
+
```
Can we find a section title for the post-TL;DR section. Maybe just:

```suggestion

## Details

```

chalin commented on content/en/blog/2024/go-opentelemetry-io-switch.md:
Consider naming this file simply `go-opentelemetry-io.md`

chalin commented on content/en/blog/2024/go-opentelemetry-io-switch.md:
```diff
@@ -0,0 +1,67 @@
+## What you need to do
+
+**There is no action required from you.** All imports using go.opentelemetry.io
+(and submodule paths, for example go.opentelemetry.io/otel, etc) will continue
```
```suggestion
(and submodule paths, for example `go.opentelemetry.io/otel`) will continue
```

chalin reviewed (commented):
Initial comments.

mx-psi reviewed (commented):
Should we clarify who is impacted by this change? I guess something like

> Downtime would be visible for users of any of the following artifacts: the OpenTelemetry Go language libraries, the OpenTelemetry Collector libraries or the OpenTelemetry Collector Builder

MrAlias commented on content/en/blog/2024/go-opentelemetry-io-switch.md:
```diff
@@ -0,0 +1,67 @@
+
+## Why are we making this change?
+
+The current app that resolves go.opentelemetry.io has not been updated in
+several years and is built on outdated dependencies. The app is currently hosted
+in a Google Cloud project that is only accessible by Google employees. Due to
+the importance of the go.opentelemetry.io domain, Google is transferring
+ownership of this app to a community-owned GCP project so that it can be updated
+and maintained by the OpenTelemetry community.
+
+For more details, see
+[the community GitHub issue proposing this change](https://github.com/open-telemetry/community/issues/2260).
```
```suggestion
This transition both helps modernize the application serving requests and handles an ownership transfer, ensuring that we are able to continue providing OpenTelemetry Go packages in a secure and robust manner.
For more details, see
[the community GitHub issue proposing this change](https://github.com/open-telemetry/community/issues/2260).
```

Going into the details here is probably not necessary.

MrAlias commented on content/en/blog/2024/go-opentelemetry-io-switch.md:
```diff
@@ -0,0 +1,67 @@
+For more details, see
+[the community GitHub issue proposing this change](https://github.com/open-telemetry/community/issues/2260).
+
+## Future plans
```
Is this section needed? Can we drop it?

dashpole replied:
+1.  I would focus on what users need to know.  Shorter is probably also better for this kind of message

chalin commented on content/en/blog/2024/go-opentelemetry-io-switch.md:
```diff
@@ -0,0 +1,67 @@
+
+## tl;dr
+
+**The app that serves requests to go.opentelemetry.io will be migrating to a new
```
No. We need to create a markdown link definition and use it where appropriate -- we probably don't want a link for _every_ occurrence (where we don't have a link, it can be in code font using backticks).

dashpole commented on content/en/blog/2024/go-opentelemetry-io-switch.md:
```diff
@@ -0,0 +1,67 @@
+**There is no action required from you.** All imports using go.opentelemetry.io
+(and submodule paths, for example go.opentelemetry.io/otel, etc) will continue
+to work. There are no plans to change this.
+
```
Where can people open issues if they are broken?  In https://github.com/open-telemetry/opentelemetry-go-vanityurls?

dashpole commented on content/en/blog/2024/go-opentelemetry-io-switch.md:
```diff
@@ -0,0 +1,67 @@
+## Why are we making this change?
+
+The current app that resolves go.opentelemetry.io has not been updated in
+several years and is built on outdated dependencies. The app is currently hosted
```
"Built on outdated dependencies" sounds a lot like "we are currently running a web service with lots of vulnerabilities".  Would it be better not to announce that?

damemi commented:
Addressed feedback, please take a look.

Spoke to @austinlparker and @MrAlias and we decided on September 9 to do the DNS switch together over a video call. Happy to invite anyone else who wants to be present to assist.

dmathieu commented on content/en/blog/2024/go-opentelemetry-io.md:
```diff
@@ -0,0 +1,84 @@
+## TL;DR
+
+The app that serves requests to [go.opentelemetry.io] will be migrating to a new
+host on September 9, 2024. There are no changes required from you. We are making
```
Should this include an approximate time? A date is very vague, especially with timezone differences.

damemi replied:
Added

dmathieu commented on content/en/blog/2024/go-opentelemetry-io.md:
```diff
@@ -0,0 +1,84 @@
+
+## What you may notice
+
+There is the possibility of the [go.opentelemtry.io] domain experiencing
```
```suggestion
There is the possibility of the [go.opentelemetry.io] domain experiencing
```

MrAlias commented on content/en/blog/2024/go-opentelemetry-io.md:
```diff
@@ -0,0 +1,83 @@
+- `go.opentelemetry.io/contrib`
+- `go.opentelemetry.io/otel`
+- `go.opentelemetry.io/proto`
+- `go.opentelemetry.io/sdk`
```
My mistake.

```suggestion
```

damemi commented:
Moved this back to draft while Tyler, Austin and I figure out a time/date that works for all 3 of us

damemi commented:
Updated the date/time and removed draft. This is ready to merge if there's no more feedback. Thanks!

tiffany76 commented on content/en/blog/2024/go-opentelemetry-io.md:
```diff
@@ -0,0 +1,82 @@
+title: Planned Migration for go.opentelemetry.io
+linkTitle: Planned migration for go.opentelemetry.io
+date: 2024-08-23
+author: >-
```
```suggestion
# prettier-ignore
cSpell:ignore: Yahn
author: >-
```

tiffany76 commented on content/en/blog/2024/go-opentelemetry-io.md:
```diff
@@ -0,0 +1,82 @@
+The app that serves requests to [go.opentelemetry.io] will be migrating to a new
+host on September 5, 2024 at approximately 11:00AM Eastern Time. There are no
+changes required from you. We are making this announcement because it is
+possible, but unlikely, that there may be downtime during this transition.
```
```suggestion
possible, but unlikely, that there might be downtime during this transition.
```

tiffany76 commented on content/en/blog/2024/go-opentelemetry-io.md:
```diff
@@ -0,0 +1,82 @@
+during this time. The OpenTelemetry engineers will be actively monitoring the
+transition to minimize the risk.
+
+### Affected Resources
```
```suggestion
### Affected resources
```

tiffany76 commented on content/en/blog/2024/go-opentelemetry-io.md:
```diff
@@ -0,0 +1,82 @@
+`go.opentelemetry.io/otel`) will continue to work. There are no plans to change
+this.
+
+## What you may notice
```
```suggestion
## What you might notice
```

tiffany76 commented on content/en/blog/2024/go-opentelemetry-io.md:
```diff
@@ -0,0 +1,82 @@
+## What you may notice
+
+There is the possibility of the [go.opentelemetry.io] domain experiencing
+downtime during this transition. If this happens, you may see projects that
```
```suggestion
downtime during this transition. If this happens, you might see projects that
```

tiffany76 reviewed (approved):
LGTM with a few small changes.

chalin commented on content/en/blog/2024/go-opentelemetry-io.md:
```diff
@@ -0,0 +1,83 @@
+---
+title: Planned Migration for go.opentelemetry.io
+linkTitle: Planned migration for go.opentelemetry.io
```
Don't repeat the title when linkTitle is identical.

```suggestion
```

chalin commented on content/en/blog/2024/go-opentelemetry-io.md:
```diff
@@ -0,0 +1,83 @@
+linkTitle: Planned migration for go.opentelemetry.io
+date: 2024-08-23
+cSpell:ignore: Yahn
+author: >-
+  [Mike Dame](https://github.com/damemi) (Google),
```
Let's drop the unnecessary trailing comma and compact it a bit:

```suggestion
author: '[Mike Dame](https://github.com/damemi) (Google)'
```

chalin commented on content/en/blog/2024/go-opentelemetry-io.md:
```diff
@@ -0,0 +1,83 @@
+cSpell:ignore: Yahn
+author: >-
+  [Mike Dame](https://github.com/damemi) (Google),
+# canonical_url: http://somewhere.else/ # TODO: if this blog post has been posted somewhere else already, uncomment & provide the canonical URL here.
+body_class: otel-with-contributions-from # TODO: remove this line if there are no secondary contributing authors
```
```suggestion
body_class: otel-with-contributions-from
```

chalin commented on content/en/blog/2024/go-opentelemetry-io.md:
```diff
@@ -0,0 +1,83 @@
+this time to minimize any downtime that occurs.
+
+If you experience any issues following this transition, please reach out on
+[Slack #otel-go](https://cloud-native.slack.com/archives/C01NPAXACKT) or open an
+issue on GitHub at
+[https://github.com/open-telemetry/opentelemetry-go-vanityurls](https://github.com/open-telemetry/opentelemetry-go-vanityurls).
```
If you're linking to the repo, you might as well link to the "new issue" action. E.g.

```suggestion
[Slack #otel-go](https://cloud-native.slack.com/archives/C01NPAXACKT) or [create an
issue](https://github.com/open-telemetry/opentelemetry-go-vanityurls/issues/new) over
[opentelemetry-go-vanityurls](https://github.com/open-telemetry/opentelemetry-go-vanityurls).
```

chalin commented on content/en/blog/2024/go-opentelemetry-io.md:
```diff
@@ -0,0 +1,83 @@
+- `go.opentelemetry.io/otel`
+- `go.opentelemetry.io/proto`
+
+Note that this includes users of any of the following artifacts: the
+OpenTelemetry Go language libraries (including Contrib), the OpenTelemetry Go
+Auto-Instrumentation libraries, the OpenTelemetry Collector libraries (including
+Contrib), and the OpenTelemetry Collector Builder.
```
How about making the "artifacts" list a bulleted list too? It would be easier to read that way.

damemi replied:
done

chalin commented on content/en/blog/2024/go-opentelemetry-io.md:
```diff
@@ -0,0 +1,83 @@
+## TL;DR
+
+The app that serves requests to [go.opentelemetry.io] will be migrating to a new
+host on September 5, 2024 at approximately 11:00AM Eastern Time. There are no
+changes required from you. We are making this announcement because it is
```
```suggestion
host on September 5, 2024 at approximately 11:00AM Eastern Time. **There are no
changes required from you**. We are making this announcement because it is
```

chalin reviewed (approved):
A few more comments, but LGTM after those are addressed. Thanks!

chalin reviewed (approved):
LGTM. I set the date to 2024-08-27. Will let whoever is the guardian of the blog gate schedule a merge, probably @svrnm or @theletterf.

dmathieu commented on content/en/blog/2024/go-opentelemetry-io.md:
```diff
@@ -0,0 +1,83 @@
+## TL;DR
+
+The app that serves requests to [go.opentelemetry.io] will be migrating to a new
+host on September 5, 2024 at approximately 11:00AM Eastern Time. **There are no
```
Nit, but IMHO, the standard timezone for communications should be GMT.

theletterf commented:
Aaaand it's live!
